### DISCUSSION TITLE
jagged_vector_buffer constructor with vecmem::vector ?

### DISCUSSION THREAD

**Discussion by beomki-yeo** [acts-project/vecmem#124] [Ideas]
@krasznaa Is it possible to replace `std::vector` arguments with `vecmem::vector` arguments?   

```
// Current constructor
jagged_vector_buffer(const std::vector<std::size_t>& sizes,
           const std::vector<std::size_t>& capacities,
           memory_resource& resource,
           memory_resource* host_access_resource = nullptr);

// What I want to have
jagged_vector_buffer(const vecmem::vector<std::size_t>& sizes,
           const vecmem::vector<std::size_t>& capacities,
           memory_resource& resource,
           memory_resource* host_access_resource = nullptr);
```

What I am trying to do is estimating capacities (in `vecmem::vector` format) from counting kernel, and put the capacities into the `jagged_vector_buffer` objects.

**Comment by krasznaa** [comment]
:thinking: I'd like to couple this with the larger cleanup, which I outlined in #95.

Unfortunately the simple solution that I first thought of (providing an optional `vecmem::copy` object to the `vecmem::data::jagged_vector_buffer` constructor), does not work. So I'll have to think about this a bit more... :frowning:

**Reply by beomki-yeo** [reply]
Okay I see. :p
